File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let map = {};
    for(let i = 0; i < strs.length; i++) {
        let word = strs[i].split("").sort().join("");
        if(map[word]) {
            map[word].push(strs[i])
        } else {
            map[word] = [strs[i]];
        }
    }
    return Object.values(map);
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        _map = {}
        for word in strs:
            sorted_word = word
            sorted_word = ''.join(sorted(word))
            if sorted_word in _map:
                _map[sorted_word].append(word)
            else:
                _map[sorted_word] = [word]
        return list(_map.values())